### PR TITLE
Feat/lints

### PR DESCRIPTION
**Pull request by saltcod**
**Project lints / issues**

https://www.notion.so/supabase/Project-Lints-f34e7b24bb5846c188c8096ad10eb045

To produce errors and warnings, run the following sql on a project: 
https://gist.github.com/saltcod/3d23dd1decb8f44c170e23f3455c3097

When these run successfully, you should see the associated warnings on 
/project/_/reports/lints

screenshot-2024-03-13-at-17 03 28

### CODE REVIEW

File: apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+    const ignoreString = currentIgnoreList.join(',')
+    setLintIgnoreList(ignoreString)
+  }
+
+  return (
+    <>
+      <Table.tr>
+        <Table.td className=" w-20">
+          <Badge
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx:52]: Badge is deprecated for <Badge_shadcn_>

File: apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          <div className="flex items-center gap-4 text-right ml-auto">
+            {lint.remediation && (
+              <Button type="text" onClick={() => setExpanded(!expanded)}>
+                <Maximize2 size={14} />
+              </Button>
+            )}
+            <DropdownMenu>
+              <DropdownMenuTrigger asChild>
+                <Button type="default" className="px-1 ml-auto">
+                  <MoreVertical size={14} />
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx:79]: size prop not needed anymore

File: apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+              <DropdownMenuContent side="bottom" className="w-[150px]">
+                <DropdownMenuItem
+                  className="flex items-center gap-2"
+                  onClick={(event) => {
+                    event.stopPropagation()
+                    setSelectedLint(lint)
+                  }}
+                >
+                  <EyeOff size={16} strokeWidth={1} />
+                  <p>
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx:91]: no <p> tag needed

File: apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            {lint.level}
+          </Badge>
+        </Table.td>
+        <Table.td className="truncate w-48">{getHumanReadableTitle(lint.name)}</Table.td>
+        <Table.td className="">{lint.description}</Table.td>
+        <Table.td className="w-16 text-right">
+          <div className="flex items-center gap-4 text-right ml-auto">
+            {lint.remediation && (
+              <Button type="text" onClick={() => setExpanded(!expanded)}>
+                <Maximize2 size={14} />
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Reports/ReportLintsTableRow.tsx:73]: size prop no longer needed

maybe add text-foreground-muted

**Comment by joshenlim** [comment]
I've got a few thoughts on the design side of things that I've written down midway - will update again on Monday! 🙏

**Comment by olirice** [comment]
Before merging it be great to add a link to each entry in the report that points to the relevant lint docs. For example, this is the page explaining what the "unindexed foreign key" lint detects, why you should care, and how to resolve https://supabase.github.io/splinter/0001_unindexed_foreign_keys/

The URL is available in the ["remediation" column of the linting query](https://github.com/supabase/splinter/blob/1d284df4c4fb6454c1b711e3a1789005e10c4cf1/splinter.sql#L44)

**Review by ivasilov — APPROVED** [review]

**Pull request merged**